packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php

```diff
@@ -41,9 +41,9 @@ public function provideData(): Iterator
     {
         foreach ($this->getDirectoriesByTagValueNodes() as $tagValueNode => $directory) {
             $filesInDirectory = $this->findFilesFromDirectory($directory);
-            foreach ($filesInDirectory as $fileInfos) {
-                foreach ($fileInfos as $fileInfo) {
-                    yield [$fileInfo, $tagValueNode];
+            foreach ($filesInDirectory as $singleFilesInDirectory) {
+                foreach ($singleFilesInDirectory as $singleSingleFilesInDirectory) {
+                    yield [$singleSingleFilesInDirectory, $tagValueNode];
```

Comment:
This should be fixer. The original foreach variable seems wrong, causing broken naming

Reply:
"filesInDirectory" when singularized with Inflector service result same value, so use "single" prefix, any other way to singularize it?

Reply:
The issue is `findFilesFromDirectory()` does not return `file[]`, but `file[][]`.

The test must be refactorted to `findFilesFromDirectory()` returns `file[]`. The inflection would work.

Another solution is to skip this file for this rule and resolve this later, as it requires more refactoring unrelated to naming.

Reply:
I will try refactor it.

Reply:
I added InflectorSingularResolver service to handle it https://github.com/rectorphp/rector/pull/5696/commits/a793125018f0f137636659b5d24279e4ae5c276a